amida-tech/amida-auth-microservice
# SER-208 / Changes that should return a 409 for duplicate entries. Also added test

If this PR fixes a bug, you _must_ add test cases representative of the bug.
#### What's this PR do?
A few changes.

Checks for a `unique violation` constraint within sequelize. If found, the error returned will be 409 (conflicts) instead of 400 (no bueno). Secondary change automatically runs fixes on `git commit` which runs `eslint watch` as part of the commit process, and modifies the integration tests to account for this (expecting 409 instead of 400). 

Two important things: One that in `user.integration.spec.js`, there's some check for "should reject updates with duplicate values" but that's not triggering the same `unique violation` that the auth endpoints are, per whatever magic sequelize is doing. Two, there is a test failing in `auth.integration.spec.js` around line 93, which is checking for the existence of a `refreshToken` it's not getting. This is there in `develop` as well, so not my monkey right this minute. 

#### Related JIRA tickets:
INBA-850. 

#### How should this be manually tested?
Run the tests first. Then, go ahead and insert a new user, confirm it goes through. Attempt to insert the same user and get a 409 back. 

#### Any background context you want to provide?
#### Screenshots (if appropriate):
